src/Factory/AppRouterFactory.php

```diff
@@ -7,14 +7,23 @@
 use Yiisoft\Router\RouterFactory;
 use Yiisoft\Yii\Web\Middleware\Controller;
 use App\Controller\SiteController;
+use Middlewares\BasicAuthentication;
+use Middlewares\DigestAuthentication;
+use App\Http\MiddlewareChain;
 
 class AppRouterFactory
 {
     public function __invoke(ContainerInterface $container)
     {
+        $basicAuth = $container->get(BasicAuthentication::class);
```

Comment:
What if I need two different username/passwords for two different URLs?

Reply:
According to [BasicAuthentication::login](https://github.com/middlewares/http-authentication/blob/a91a5909ec6be91192dbfaccc487afcd434be58a/src/BasicAuthentication.php#L48), it seems impossible to do that, unless extends and overwrite it, or we can implement our own middleware.

Reply:
I don't think it is impossible. Likely, using yiisoft/factory or simply using `new` instead of getting `BasicAuthentication` from container for these would allow setting up multiple instances of `BasicAuthentication` with different `users` constructor parameter.